.github/workflows/collector-builder.yml

```diff
@@ -74,6 +75,24 @@ jobs:
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v3
 
+      - name: Set up builder ccache
+        if: env.USE_CCACHE
+        uses: actions/cache@v4
+        with:
+          path: ${{ github.workspace }}/builder/.ccache
+          key: builder-ccache-${{ matrix.arch }}-${{ github.sha }}
+          restore-keys: |
+            builder-ccache-${{ matrix.arch }}-
+
+      - name: Set up builder ccache in docker cache
+        if: env.USE_CCACHE && matrix.arch != 's390x'
+        uses: reproducible-containers/buildkit-cache-dance@v3.1.2
+        with:
+          cache-map: |
+            {
+              "${{ github.workspace }}/builder/.ccache": "/root/.ccache"
+            }
```

Comment:
Since you had to implement the logic that this action provides for s390x, I wonder if it makes sense to just handle all archs the same way in our ansible playbooks and drop this step, mostly for consistency, this approach is fine.